### ISSUE TITLE
[Bug]: 转写失败

### ISSUE THREAD

**Issue by foreverlong** [QuintinShaw/openasr#159]
### OpenASR version or commit SHA

0.1.17桌面版

### OS

Windows10

### GPU model and graphics driver version

gtx 1660ti

### Command run or API request

```shell
桌面端
```

### Expected behavior

转写成功

### Actual behavior

Daemon returned HTTP 400.

Could not transcribe audio: Native ASR Core transcription stayed fail-closed after local runtime source validation/dispatch: Native ASR session failed: Native ASR Core transcription stayed fail-closed after local runtime source validation/dispatch: ggml executor 'qwen3-asr-ggml-executor-v1' failed for adapter 'ggml-family-qwen3-asr-runtime-v1': qwen3-asr greedy decode loop failed: qwen3-asr greedy decode decoder step failed: qwen3-asr greedy decode decoder step failed: qwen3-asr llm logits head ggml graph failed: ggml cpu graph backend buffer allocation failed
No partial transcript was emitted..
No partial transcript was emitted.

### Logs or output

```text

```

### Was OPENASR_HOME customized?

No

### Backend used

Not sure / did not pass --backend (this is native, the default)

### Native .oasr pack details, if used

Qwen3-ASR 1.7B

### Audio sharing confirmation

- [x] I have not attached private audio unless it is safe to share publicly.

**Comment by QuintinShaw** [comment]
这个报错是引擎在内存不够时的主动安全停止（转写 Qwen3-ASR 1.7B 需要较多内存,词表输出层的计算缓冲分配失败了），文件本身没问题。

帮我们定位，麻烦补充：
1. 电脑内存多大（这个错误和物理内存关系最大）;
2. daemon.log 和 desktop.log（高级设置设置 → 打开日志文件夹）；
3. 音频多长。

临时解决：试试 Qwen3-ASR 的 q4/q8 量化版（模型市场里同名模型选更小的版本），或换 SenseVoice/X-ASR 这类轻量模型；转写前关闭占内存大的程序也有帮助。

**Comment by foreverlong** [comment]
使用SenseVoice成功了，非常感谢。
以及能否接入funASR？funASR有发言人、时间轴等等，感觉还是比较准确的，只是苦于没有好用的gui，如果能集成进来就好了。
from funasr import AutoModel

model = AutoModel(
    model="paraformer-zh",          # 中文 ASR
    vad_model="fsmn-vad",           # 支持任意长度音频
    vad_kwargs={"max_single_segment_time": 60000},
    punc_model="ct-punc",           # 添加标点
    spk_model="cam++",
)
#res = model.generate(input="video.mp4",batch_size_s=300)
res = model.generate(input="recorder.wav",batch_size_s=300)

for sent in res[0]["sentence_info"]:
    print(f"[说话人 {sent['spk']}] [{sent['start']}-{sent['end']}ms] {sent['text']}")

**Comment by QuintinShaw** [comment]
感谢你之前提供的信息，0.1.19桌面版（内核0.1.23）已经发布了，目前：

- 显存不够时自动降级到 CPU — 遇到显存不足，会自动回退用 CPU 跑完，不再整个任务直接失败。

因为我测试使用的短音频，可能内存占用不高，你可以尝试选择 0.6B 的 Qwen3 ASR，理论上 1.7B 的内存占用是支持的，稍后我会继续针对 Qwen3 ASR 系列的内存做持续优化工作，此外 FireRedASR2 AED-L 这个模型昨天刚优化完成，速度很快，也是中文识别的 SOTA，可以体验一下，也欢迎随时反馈。FunASR 我会在 Moss 这个模型接入后接入，预计本周内，同时也会完善说话人分离这些功能并拓展到全部模型上，这个 issue 可以先保留，我会在 FunASR 接入时关闭的

**Comment by foreverlong** [comment]
大佬好，又遇到一个转写问题，我就不新开issue了。
Daemon returned HTTP 400.

Could not transcribe audio: Native ASR Core transcription stayed fail-closed after local runtime source validation/dispatch: Native ASR session failed: Native ASR Core input format is unsupported for local inference: Native ASR session failed: expected 16 kHz mono PCM16 or float32 WAV input for Native ASR Core backend.
Provide 16 kHz mono PCM WAV input (or normalize before backend dispatch)..
No partial transcript was emitted.
我查着是因为格式不行？是因为不支持双声道吗？谢谢。

**Comment by LauraGPT** [comment]
这个错误已经把约束写得比较明确：Native ASR Core 目前只接受 16 kHz、单声道、PCM16 或 float32 的 WAV。双声道会触发，但 44.1/48 kHz、压缩 WAV（如 ADPCM）也会触发，所以不能只看扩展名。

可以先生成一个新文件，不覆盖原音频：

```bash
ffmpeg -i recorder.wav -map 0:a:0 -ac 1 -ar 16000 -c:a pcm_s16le recorder_16k_mono.wav
```

然后确认输出契约：

```bash
ffprobe -v error -select_streams a:0 -show_entries stream=codec_name,sample_rate,channels -of default=noprint_wrappers=1 recorder_16k_mono.wav
```

预期看到 `codec_name=pcm_s16le`、`sample_rate=16000`、`channels=1`。再把 `recorder_16k_mono.wav` 交给桌面端即可。

从产品行为看，桌面端后续最好在进入 native backend 前自动做重采样/下混，并在失败信息里列出实际 codec、采样率和声道数；这样用户选择文件后不需要手工理解音频容器。FunASR 的 Python 文件输入路径通常会承担解码和重采样，但当前 OpenASR native core 的严格输入契约仍需先满足。
